Include/Geometry/strided_reads_gpu.h

```diff
@@ -0,0 +1,180 @@
+#ifndef STRIDED_READS_GPU_HPP
+#define STRIDED_READS_GPU_HPP
+
+#ifdef FIXED_STRIDE
+    #define THREADSIZE 32
+#else
+    #define THREADSIZE 1
+#endif
+
+//#include "libhr_core.h"
+#include "geometry.h"
+
+#ifdef __cplusplus
```

Comment:
These are the only gpu read/write functions that are used at the moment. They are directly used kernels, such as the dirac operator kernels, syncing the field to the sendbuffer in the new geometry and linear algebra. The reading and writing functions in the template that can also be used in C code actually call these C++ functions. 

I am planning to remove the old generated versions in Include/Geometry/strided_reads.h but was reluctant to do it now, because of the amount of changes that I already made.

Reply:
It'd be better to move the templated c++ implementation of the functions to a separate hpp file in LibHR, and only keep the c functions in this header